TetrisEngine.java

```diff
@@ -0,0 +1,134 @@
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+
+public class TetrisEngine {
+	private static final String INPUT_FILE = "input.txt";
+	private static final String OUTPUT_FILE = "output.txt";
+	//private static final int GRID_SIZE = 100;
+	
+	private static HashMap<Character, TetrisPiece> PieceMap = new HashMap<Character, TetrisPiece>();
+	private static TetrisGrid Grid = new TetrisGrid();
+	
+	public static void main(String[] args) {
+		PopulatePieceMap();
+		List<String> inputLines = ReadFile();
+		String output = "";
+
+		for(String line : inputLines)
+		{
+			line = line.trim();
+			output += DoTetris(line);
+		}
+		WriteFile(output);
+	}
+	
+	private static String DoTetris(String input)
+	{
+		String out = "";
+		Grid.Reset();
+		for(String shapeStr : input.split(","))
+		{			
+			if(shapeStr.length() != 2)
+				return "Invalid Input";
+			
+			Grid.AddPiece(MakePiece(shapeStr.charAt(0), Character.getNumericValue(shapeStr.charAt(1))));
+		}
+		Grid.toString();
+		out = String.format("%s\n", Grid.getHeight());
+		return out;
+	}
+	
+	private static TetrisPiece MakePiece(char shape, int index)
+	{		
+		TetrisPiece newPiece = new TetrisPiece(PieceMap.get(shape), index);				
+		return newPiece;
+	}
+	
+	private static List<String> ReadFile()
+	{
+		List<String> input = null;
+		try {
+			
+			input = Files.readAllLines(Paths.get(INPUT_FILE));
+
+		} catch (IOException e) {
+

```

Comment:
Your program should fail if an IOException is thrown, since it won't be able to read the input file. Suggest printing the error to the stderr here:
System.err.println(e);
e.printStackTrace();